pkg/target/manager.go

```diff
@@ -567,3 +603,29 @@ func writeRawConfig(config clientcmd.ClientConfig) ([]byte, error) {
 
 	return clientcmd.Write(rawConfig)
 }
+
+func getProjectNamespace(ctx context.Context, client gardenclient.Client, name string) (*string, error) {
+	project, err := client.GetProject(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+
+	if project.Spec.Namespace == nil || *project.Spec.Namespace == "" {
+		return nil, fmt.Errorf("project %q has not yet been assigned to a namespace", name)
+	}
+
+	return project.Spec.Namespace, nil
+}
+
+func clientConfigWithNamespace(clientConfig clientcmd.ClientConfig, namespace string) (clientcmd.ClientConfig, error) {
+	rawConfig, err := clientConfig.RawConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get raw client config: %w", err)
+	}
+
+	contextName := rawConfig.CurrentContext
+	rawConfig.Contexts[contextName].Namespace = namespace
```

Comment:
check if exists (rawConfig.Contexts[contextName])

Reply:
✔️

Reply:
We now mimify the config and set the contextName on all contexts (e.g. internal and external) belonging to the current cluster.